src/login/logind-session-device.c

```diff
@@ -367,9 +359,39 @@ int session_device_new(Session *s, dev_t dev, SessionDevice **out) {
         r = hashmap_put(s->devices, &sd->dev, sd);
         if (r < 0) {
                 r = -ENOMEM;
+                free(sd->node);
```

Comment:
this looks really strange, you need to reset sd->node to NULL here, otherwise it will point into invalidated memory? also, why reset that here anyway? I am pretty sure functions should try to rollback only the changes they made themselves on failure, but this function never initializes node